### ISSUE TITLE
Bag Issues possible bug

### ISSUE THREAD

**Issue by Kobra299** [NetroScript/Graveyard-Keeper-Savefile-Editor#71]
I have noticed a issue with save files when you have a couple of bags in your inventory if you edit the save it moves all the items from one bag to another and deletes the items in the second bag

**Comment by NetroScript** [comment]
Can you supply me a save file where that happens with both a before and after version?
(Upload both in a zipped file)

**Comment by Kobra299** [comment]
I have the 2 save files the one before and the one after I used your save editor to add some bee wax and deleting the hops_crop3 in my main invintory it has done a secondary glitch of making the second univeral bag none working this has happened as a alt to the main glitch
[After Glitch.zip](https://github.com/NetroScript/Graveyard-Keeper-Savefile-Editor/files/5639883/After.Glitch.zip)
[Before Glitch.zip](https://github.com/NetroScript/Graveyard-Keeper-Savefile-Editor/files/5639884/Before.Glitch.zip)

**Comment by Kobra299** [comment]
any update to this

**Comment by NetroScript** [comment]
I took a look at it now.

The way saving items works is relatively simple because it wasn't needed in a complex way in the beginning.

It first fits the items array to the correct length (bei either removing instances or adding new ones).

Then it assigns the main meta data to those items (meaning for example the ID and the durability).

So if you have 2 items, 1 is a bag and 1 is any other random item, and for some reason they change places the other item will have the ID of a bag but not its meta data, and the original bag will have the meta data of a bag, but will be an item which won't make use of it (I guess the game will then discard that additional data).

I maybe do have ideas to fix that, but problems would remain like if you had a new bag to the inventory, the save editor would have no way to know the meta data of a bag, unless it had it stored itself somewhere and copies it over.

**Comment by Kobra299** [comment]
ok thats good to known and thank you for looking in to fixing this was not sure if it was a bug or not hence the post

**Comment by NetroScript** [comment]
Like I said I will see and try to improve it (if it works out like I intend it to).
In any case I will also add at least item view for sub inventories (like the universal bag) and that already shows clearly what the problem is (and how it just duplicated the items, which I will now try to make a bit smarter).

On the left the before save, on the right the after save (where you can see both are already a bit broken because of the way the editor juggles the item data) luckily it still works ingame (because it just ignores that data while loading I guess).

![Screenshot of the 2 different saves](https://user-images.githubusercontent.com/18115780/102719686-af43b880-42ef-11eb-8d76-3a39f0b434e3.png)

**Comment by NetroScript** [comment]
I also investigated why your specific backpack stopped working, the problem there is it copied over the meta data for the teleport stone (which is the cooldown it has) and overwrote the inventory size (which is now non existant). So you could add functionality back by doing a manual edit, but it would be pretty meh to do by hand (if you ever wanted to do that, there I would suggest using the HTML JSON export)

**Comment by Kobra299** [comment]
its cool I can just make a new bag and get a tp stone at the pub so this fine if you want to leave this open while you do the test for the update it cool thank you for helping me with this

**Comment by NetroScript** [comment]
I already finished most of the code today, I will only have to clean it up / finish additional files tomorrow.

The next version of the editor will include better inventory handling by quite a bit. Well in the GUI it will look the same. Under the hood it reuses items and uses fallback items. It also allows special cases (like the bags) to be added to the inventory without being broken, but for that I need saves with working instances of special case items. I personally don't know which items all have custom meta data, in the version I will publish next I have included `universal_bag` and `universal_bag_big` which I just took from your save.
